test/unit/pipeline/FetchUnitTest.cc

```diff
@@ -125,7 +131,8 @@ TEST_F(PipelineFetchUnitTest, FetchUnaligned) {
       .WillByDefault(DoAll(SetArgReferee<3>(mOp), Return(4)));
```

Comment:
ON_CALL for predecode should return min or max instruction size, with the following EXPECT_CALLs also altered so that the number of times each is expected is based off of the return value or predecode (where relevant)

Reply:
This will break the below (131-134) as we then can't guarantee how many instructions will be in the buffer e.g. if max is 6. This needs to be updated with the TODO